src/tblib/__init__.py

```diff
@@ -53,21 +53,38 @@ def __init__(self, frame):
 
 
 class Traceback(object):
+
+    tb_next = None
+
     def __init__(self, tb):
         self.tb_frame = Frame(tb.tb_frame)
         # noinspection SpellCheckingInspection
         self.tb_lineno = int(tb.tb_lineno)
-        if tb.tb_next is None:
-            self.tb_next = None
-        else:
-            self.tb_next = Traceback(tb.tb_next)
+
+        # Build in place to avoid exceeding the recursion limit
+        tb = tb.tb_next
+        tbs = [self]
+        cls = type(self)
+        while tb is not None:
+            traceback = object.__new__(cls)
+            traceback.tb_frame = Frame(tb.tb_frame)
+            traceback.tb_lineno = int(tb.tb_lineno)
+            tbs[-1].tb_next = traceback
+            tbs.append(traceback)
```

Comment:
It's not really necessary to keep all the objects in the list, only the last one right? Not a problem from GC perspective since there will be a reference to all the objects (from the previous one) except the last (for which we would have a local variable)